pkg/guestagent/events/containerd_linux.go

```diff
@@ -0,0 +1,365 @@
+// SPDX-FileCopyrightText: Copyright The Lima Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package events
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/api/events"
+	"github.com/containerd/containerd/errdefs"
+	containerdNamespace "github.com/containerd/containerd/namespaces"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/lima-vm/lima/v2/pkg/guestagent/api"
+)
+
+const (
+	stateKey             = "nerdctl/state-dir"
+	portsKey             = "nerdctl/ports"
+	namespaceKey         = "nerdctl/namespace"
+	defaultSocketTimeout = 5 * time.Second
+)
+
+type ipPortMap map[string][]*api.IPPort
```

Comment:
We should have a type for `[]*api.IPPort` as well because it is used as a parameter / return type multiple times. And let's use plural because it is an array of values:

```suggestion
type ipPorts []*api.IPPort
type ipPortsMap map[string]ipPorts
```

Reply:
I think introducing a named type here adds an unnecessary level of abstraction. I prefer keeping `[]*api.IPPort` as-is, since it's immediately clear and self-explanatory in the context of the code what type we are referring to. Using a custom alias makes it harder to quickly understand what type we're working with—especially when it’s just a generic slice without domain-specific behavior.

This kind of abstraction also feels like premature optimization for reuse. In many standard Go libraries and widely-used codebases, slice types—even when reused—are typically kept inline unless they encapsulate behavior or have domain meaning.

Reply:
Ok, fine. I guess it just looks like it wants a type definition to me because it is a slice of pointers.